xdsl/dialects/onnx.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from typing import Annotated, TypeVar
+
+from xdsl.dialects.builtin import (
+    SSAValue,
+    TensorType,
+)
+from xdsl.ir import (
+    Attribute,
+    Dialect,
+)
+from xdsl.irdl import (
+    ConstraintVar,
+    IRDLOperation,
+    irdl_op_definition,
+    operand_def,
+    result_def,
+)
+from xdsl.utils.exceptions import VerifyException
+
+_T = TypeVar("_T", bound=Attribute)
+
+
+class ElementwiseBinOpBase(IRDLOperation):
```

Comment:
```suggestion
class ElementwiseBinOpBase(IRDLOperation, ABC):
```
To be sure we don't make the mistake of instantiating it directly.